### PR TITLE
Feature/s16 francis 30

### PR DESCRIPTION
**Pull request by kumasuje**
Hi Francis,

This feature includes git Push functionality with token based authentication.  

Note : Required Exception handling and logging will be merged in feature 35 and 36.

Regards,
Sujeet Kumar

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,96 @@
 			<version>2.4.4</version>
 		</dependency>
 
-	</dependencies>
+	 <!-- Dependencies for Git Module <added by Sujeet, starting from here>  -->
+    
+    <dependency>
```
Comment (burnumd) [inline pom.xml]: You should probably, as a team, come up with an indentation standard (I think four spaces is typical for pom.xml).

File: src/main/java/org/p632/turnkey/beans/ProjectBuilderBean.java [hunk 2]
```diff
@@ -20,24 +19,21 @@
...
-		{
+
+		try {
 			ClassLoader classLoader = getClass().getClassLoader();
-			File xmlFile = new File(classLoader.getResource("dependency.xml").getFile());			
+			File xmlFile = new File(classLoader.getResource("dependency.xml").getFile());
 			DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
 			DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
-			Document doc = dBuilder.parse(xmlFile);		
+			Document doc = dBuilder.parse(xmlFile);
```
Comment (burnumd) [inline src/main/java/org/p632/turnkey/beans/ProjectBuilderBean.java:30]: In the future you might save yourselves some headaches by looking into something like https://github.com/FasterXML/jackson-dataformat-xml

File: .gitignore [hunk 3]
```diff
@@ -13,8 +14,6 @@ libs
 # Classpath
 .classpath
 
-# Config properties
-config.properties
```
Comment (burnumd) [inline .gitignore:17]: Why are you no longer ignoring the config.properties? If it contains your auth token, it shouldn't be in VCS.

File: src/main/java/org/p632/turnkey/beans/ProjectBuilderBean.java (hunk 2 above)
Comment (rohsurve) [inline src/main/java/org/p632/turnkey/beans/ProjectBuilderBean.java:30]: Handled in Issue  S16FRANCIS-40

**Comment by rohsurve** [comment]
Issues such as xml parsing is handled in  S16FRANCIS-40  and Externalizing configuration  && exception and logging configuration is  S16FRANCIS-35.

**Comment by akshaykamath** [comment]
The issues raised above are fixed in the mentioned branches.

**Pull request merged**